src/DashboardService.Business/TaskType/GetTaskTypesCommand.cs

```diff
@@ -1,25 +1,39 @@
 using LT.DigitalOffice.DashboardService.Business.TaskType.Interfaces;
 using LT.DigitalOffice.DashboardService.Data.Interfaces;
+using LT.DigitalOffice.DashboardService.Mappers.Models.Interfaces;
+using LT.DigitalOffice.DashboardService.Models.Db;
 using LT.DigitalOffice.DashboardService.Models.Dto.Models;
 using LT.DigitalOffice.DashboardService.Models.Dto.Requests.TaskType.Filters;
 using LT.DigitalOffice.Kernel.Responses;
 using Microsoft.AspNetCore.Mvc;
 using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace LT.DigitalOffice.DashboardService.Business.TaskType;
 
 public class GetTaskTypesCommand : IGetTaskTypesCommand
 {
-  private readonly ITaskTypeRepository _repository;
-  
-  public GetTaskTypesCommand(ITaskTypeRepository repository) 
+  private readonly ITaskTypeRepository _taskTypeRepository;
+  private readonly ITaskTypeInfoMapper _taskTypeInfoMapper;
+
+  public GetTaskTypesCommand(
+    ITaskTypeRepository repository,
+    ITaskTypeInfoMapper taskTypeInfoMapper)
   {
-    _repository = repository;
+    _taskTypeRepository = repository;
+    _taskTypeInfoMapper = taskTypeInfoMapper;
   }
   
-  public Task<FindResultResponse<TaskTypeInfo>> ExecuteAsync(GetTaskTypesFilter filter)
+  public async Task<FindResultResponse<TaskTypeInfo>> ExecuteAsync(GetTaskTypesFilter filter, CancellationToken ct)
   {
-    throw new System.NotImplementedException();
+    (List<DbTaskType> dbTaskTypes, int totalCount) result = await _taskTypeRepository.GetAllAsync(filter, ct);
+
+    return new()
+    {
+      Body = result.dbTaskTypes.Select(p => _taskTypeInfoMapper.Map(p)).ToList(),
```

Comment:
May be change p => _taskTypeInfoMapper.Map(p) on _taskTypeInfoMapper.Map?

Reply:
++